simulator/src/main/java/kcl/teamIndexZero/traffic/simulator/data/trafficLight/TrafficLightSet.java

```diff
@@ -10,7 +18,92 @@
 
 public class TrafficLightSet implements ISimulationAware {
     private ID id;
+    public List<TrafficLightInSet> TrafficLightSetList;
+    public List<TrafficLightInSet> InteriorListA;
+    public List<TrafficLightInSet> InteriorListB;
+    private static Logger_Interface LOG = Logger.getLoggerInstance(TrafficLightSet.class.getSimpleName());
+
+    /**
+     * Constructor
+     *
+     * @param id TrafficLightSet ID
+     */
+    public TrafficLightSet(ID id) {
+        this.id=id;
+        this.TrafficLightSetList =new ArrayList<TrafficLightInSet>();
+        this.InteriorListA =new ArrayList<TrafficLightInSet>();
+        this.InteriorListB =new ArrayList<TrafficLightInSet>();
+    }
+
+    /**
+     * Adds traffic lights to the List of traffic lights within one junction
+     *
+     * @param trafficLightLnSet object to be added to the list
+     */
+    public void addTrafficlight(TrafficLightInSet trafficLightLnSet, char type ){ // type: A or B, the type of the InteriorList
```

Comment:
Okay, makes sense - totally agree at that point. 
If we still happen to have time to merge this down ,we will need to see how to make things work. 

There is a slight problem with 'opposite' notion in real-life model: for the usual + shape crossings, we have 4 roads connected to a crossing. But the problem is, it is really non-straightforward to say what is 'opposite'. Say, the road to north has an angle of 90 degrees to east. And the road to south has something not 270˚ as you'd expect, but, say, 266˚ - totally usual thing in real data. 

So in order to identify what's opposite we'll need to do some extra work in junction construction. Code for determining angles is there - that's group of `getBearing`, `getBearingForLane` etc methods. But judging on what is what using these angles is an extra thing to keep in mind